File: visa-dev-samplecode-master/vdp-java/src/test/java/com/visa/vdp/vta/TestManagePortfolios.java
package com.visa.vdp.vta;

import java.util.HashMap;
import java.util.Map;

import org.apache.http.client.methods.CloseableHttpResponse;
import org.testng.Assert;
import org.apache.http.HttpStatus;
import org.testng.annotations.Test;

import com.visa.vdp.utils.VisaAPIClient;
import com.visa.vdp.utils.MethodTypes;
import com.visa.vdp.utils.Property;
import com.visa.vdp.utils.VisaProperties;

public class TestManagePortfolios {

    VisaAPIClient visaAPIClient;

    @Test(groups = "vta")
    public void setUp() {
        this.visaAPIClient = new VisaAPIClient();
    }

    @Test(groups = "vta")
    public void testPortfolios() throws Exception {
        String baseUri = "vta/";
        String resourcePath = "v3/communities/"+VisaProperties.getProperty(Property.VTA_COMMUNITY_CODE) +"/portfolios";

        Map<String,String> headers = new HashMap<String,String>();
        headers.put("ServiceId", VisaProperties.getProperty(Property.VTA_SERVICE_ID));
        CloseableHttpResponse response = this.visaAPIClient.doMutualAuthRequest(baseUri + resourcePath, "Get Portfolio Details Test", "", MethodTypes.GET, headers);
        Assert.assertEquals(response.getStatusLine().getStatusCode(), HttpStatus.SC_OK);
        response.close();
    }
}

File: vdp-python/visa/test/vta/test_manage_portfolios.py
from visa.helpers.visa_api_client import VisaAPIClient
import unittest
import sys
import os
if sys.version_info < (3, 0):
    import ConfigParser as parser
else:
    import configparser as parser
'''
@author: visa
'''

class TestManagePortfolios(unittest.TestCase):
    
    config = parser.ConfigParser()
    config_path = os.path.abspath(os.path.join(os.path.dirname(os.path.dirname(__file__)),'..','configuration.ini'))
    config.read(config_path)
    
    def setUp(self):
        self.visa_api_client = VisaAPIClient()
    
    def test_portfolios(self):
        base_uri = 'vta/'
        resource_path = 'v3/communities/'+ self.config.get('VDP','vtaCommunityCode') +'/portfolios'
        response = self.visa_api_client.do_mutual_auth_request(base_uri + resource_path, '', 'Get Portfolio Details Test', 'get', {'ServiceId' : self.config.get('VDP','vtaServiceId')})
        self.assertEqual(str(response.status_code) ,"200" ,"Get Portfolio Details Test failed")
        pass
